pkg/generator/fluentd/output/syslog/syslog.go

```diff
@@ -0,0 +1,395 @@
+			TemplateName: "syslogParseJson",
+			TemplateStr: `{{define "syslogParseJson" -}}
+@type parse_json_field
+json_fields  message
+merge_json_log false
+replace_json_log true
+{{end}}`,
+		},
+	}
+}
+
+func AddLogSource(o logging.OutputSpec, op Options) Element {
+	return Filter{
+		MatchTags: "**",
+		Element: RecordModifier{
+			Records: []Record{
+				{
+					Key:        "kubernetes_info",
+					Expression: `${if record.has_key?('kubernetes'); record['kubernetes']; else {}; end}`,
+				},
+				{
+					Key:        "namespace_info",
+					Expression: `${if record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; "namespace_name=" + record['kubernetes_info']['namespace_name']; else nil; end}`,
+				},
+				{
+					Key:        "pod_info",
+					Expression: `${if record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; "pod_name=" + record['kubernetes_info']['pod_name']; else nil; end}`,
+				},
+				{
+					Key:        "container_info",
+					Expression: `${if record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; "container_name=" + record['kubernetes_info']['container_name']; else nil; end}`,
+				},
+				{
+					Key:        "msg_key",
+					Expression: `${if record.has_key?('message') && record['message'] != nil; record['message']; else nil; end}`,
+				},
+				{
+					Key:        "msg_info",
+					Expression: `${if record['msg_key'] != nil && record['msg_key'].is_a?(Hash); require 'json'; "message="+record['message'].to_json; elsif record['msg_key'] != nil; "message="+record['message']; else nil; end}`,
+				},
+				{
+					Key:        "message",
+					Expression: `${if record['msg_key'] != nil && record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; record['namespace_info'] + ", " + record['container_info'] + ", " + record['pod_info'] + ", " + record['msg_info']; else record['message']; end}`,
+				},
+				{
+					Key:        "systemd_info",
+					Expression: `${if record.has_key?('systemd') && record['systemd']['t'].has_key?('PID'); record['systemd']['u']['SYSLOG_IDENTIFIER'] += "[" + record['systemd']['t']['PID'] + "]"; else {}; end}`,
+				},
+			},
+			RemoveKeys: []string{
+				"kubernetes_info",
+				"namespace_info",
+				"pod_info",
+				"container_info",
+				"msg_key",
+				"msg_info",
+				"systemd_info",
+			},
+		},
+	}
+}
+
+func SecurityConfig(o logging.OutputSpec, secret *corev1.Secret) []Element {
+	if o.Secret != nil {
+		conf := []Element{
+			TLS(true),
+		}
+		if secret != nil && security.HasCABundle(secret) {
+			ca := CAFile{
+				// TODO: use constants.TrustedCABundleKey
+				CAFilePath: security.SecretPath(o.Secret.Name, "ca-bundle.crt"),
+			}
+			conf = append(conf, ca)
+		}
+		return conf
+	} else {
+		return nil
+	}
+}
+
+// The Syslog output fields can be set to an expression of the form $.abc.xyz
+// If an expression is used, its value will be taken from corresponding key in the record
+var keyre = regexp.MustCompile(`^\$(\.[[:word:]]*)+$`)
```

Comment:
can we add one example of key and tag here?

Reply:
Added